### PR TITLE
added solution method

### PR DESCRIPTION
**Pull request by tsehelnyk**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
+        this.model = model;
+        this.color = color;
+        this.bodyType = bodyType;
+        this.win = win;
+        this.yearOfIssue = yearOfIssue;
+    }
+
+    @Override
+    public Car clone() {
+        Car clonedCar = new Car(this.brand, this.model, this.color,
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: remove all `this` keywords, please

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+                this.bodyType, this.win, this.yearOfIssue);
+        clonedCar.engine = engine.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: try to use `engine.clone()` in the Car constructor

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
+        this.type = type;
+        this.fuel = fuel;
+        this.capacity = capacity;
+        this.cylinders = cylinders;
+        this.serialNumber = serialNumber;
+    }
+
+    @Override
+    public Engine clone() {
+        Engine clonedEngine = new Engine(this.type, this.fuel, this.capacity,
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]: let's remove all `this` keywords

File: src/main/java/core/basesyntax/Engine.java [hunk 4]
```diff
@@ -1,5 +1,26 @@
...
+                this.cylinders, this.serialNumber);
+        return clonedEngine;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]:
```suggestion
        return new Engine(type, fuel, ...);
```

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (tsehelnyk) [reply src/main/java/core/basesyntax/Car.java]: ok

File: src/main/java/core/basesyntax/Car.java (hunk 2 above)
Comment (tsehelnyk) [reply src/main/java/core/basesyntax/Car.java]: ok

File: src/main/java/core/basesyntax/Engine.java (hunk 3 above)
Comment (tsehelnyk) [reply src/main/java/core/basesyntax/Engine.java]: ok

File: src/main/java/core/basesyntax/Engine.java (hunk 4 above)
Comment (tsehelnyk) [reply src/main/java/core/basesyntax/Engine.java]: ready

**Pull request closed without merging**